### PR TITLE
JAVA-5505 minor refactoring

### PR DESCRIPTION
**Pull request by katcharov**
JAVA-5505

Mostly-automated refactoring to remove IDE nags. I've left unused methods like `getCurrentRegistrationCount` in `AsynchronousTlsChannelGroup`.

### CODE REVIEW

**Review by stIncMale — COMMENTED** [review]
I am not sure it's wise to refactor `tlschannel` code, which at some point we may have to synchronize with the upstream again. Detecting which discrepancies are functional and must not be lost when we synchronize, and which were done for other reasons, may be more difficult if we change the `tlschannel` code without necessity.

File: driver-core/src/main/com/mongodb/internal/connection/tlschannel/async/AsynchronousTlsChannelGroup.java [hunk 1]
```diff
@@ -159,7 +159,7 @@ static final class WriteOperation extends Operation {
 
     private final Selector selector;
 
-    final ExecutorService executor;
+    private final ExecutorService executor;
```
Comment (katcharov) [inline driver-core/src/main/com/mongodb/internal/connection/tlschannel/async/AsynchronousTlsChannelGroup.java:162]: @stIncMale I can roll back the finals. The reduced visibility refactor would ensure that (in the near future) the user-set executor is not shut down from outside this class. Up to you - is this worth the potential future merge conflict?

Comment (stIncMale) [reply driver-core/src/main/com/mongodb/internal/connection/tlschannel/async/AsynchronousTlsChannelGroup.java:162]: This looks like something worth discussing the team on the catchup meeting, or asking Jeff/Ross here, as they are likely the only ones who have idea on our process of syncing the `tlschannel` code.

Comment (katcharov) [reply driver-core/src/main/com/mongodb/internal/connection/tlschannel/async/AsynchronousTlsChannelGroup.java:162]: Force rebased. Removed finals commit, kept visibility commit.

File: driver-core/src/main/com/mongodb/internal/connection/tlschannel/async/AsynchronousTlsChannelGroup.java [hunk 2]
```diff
@@ -702,7 +706,7 @@ public void shutdownNow() {
      *
      * @return whether the channel is terminated
      */
-    public boolean isTerminated() {
+    private boolean isTerminated() {
```
Comment (stIncMale) [inline driver-core/src/main/com/mongodb/internal/connection/tlschannel/async/AsynchronousTlsChannelGroup.java:709]: This change seems irrelevant to making sure that the rest of the `tlschannel` code can't terminate a user-supplied executor. If that's so, let's not do the change.

**Review by stIncMale — CHANGES_REQUESTED** [review]

**Review by stIncMale — APPROVED** [review]

**Comment by katcharov** [comment]
Test failures appear to be unrelated.
